activitysim/core/chunk.py

```diff
@@ -1039,16 +1046,79 @@ def ledger(self):
                 CHUNK_LEDGERS.pop()
                 self.chunk_ledger = None
 
+    def log_rss(self, trace_label, force=False):
+        if self.chunk_training_mode == MODE_CHUNKLESS:
+            # no memory tracing at all in chunkless mode
+            return
+
+        assert len(CHUNK_LEDGERS) > 0, f"log_rss called without current chunker."
+
+        hwm_trace_label = f"{trace_label}.log_rss"
+
+        if self.chunk_training_mode == MODE_PRODUCTION:
+            # FIXME - this trace_memory_info call slows things down a lot so it is turned off for now
+            # trace_ticks = 0 if force else mem.MEM_TRACE_TICK_LEN
+            # mem.trace_memory_info(hwm_trace_label, trace_ticks=trace_ticks)
+            return
+
+        rss, uss = mem.trace_memory_info(hwm_trace_label, state=self.state)
+
+        # check local hwm for all ledgers
+        with ledger_lock:
+            for c in CHUNK_LEDGERS:
+                c.check_local_hwm(hwm_trace_label, rss, uss, total_bytes=None)
```

Comment:
missing state argument?
```suggestion
                c.check_local_hwm(self.state, hwm_trace_label, rss, uss, total_bytes=None)
```